solr/core/src/java/org/apache/solr/handler/component/HttpShardHandler.java

```diff
@@ -62,7 +69,14 @@ public class HttpShardHandler extends ShardHandler {
   private HttpShardHandlerFactory httpShardHandlerFactory;
   private Map<ShardResponse, CompletableFuture<LBSolrClient.Rsp>> responseFutureMap;
```

Comment:
It seems responseFutureMap is guarded by RESPONSE_CANCELABLE_LOCK; could use a comment to say so